resources/js/common.js

```diff
@@ -73,91 +71,68 @@ window.onload = (event) => {
 
   // Blog post section
   let postLists = document.querySelector('.posts')
-  utils.jsonCaller('get','resources/json/blogpost.json', function(blogObj){
-     for(let blog of blogObj){
-       let blogPost = createPost(blog.tittle, blog.image)
-       postLists.appendChild(blogPost)
-     }
+  utils.jsonCaller('get', 'resources/json/blogpost.json', function (blogObj) {
+    for (let blog of blogObj) {
+      let blogPost = createPost(blog.tittle, blog.image)
+      postLists.appendChild(blogPost)
+    }
   })
 
 
   function createPost(tittle, imgLink) {
-    const postElement = document.createElement('li');
-    const postLink = document.createElement('a');
+    const postElement = document.createElement('li') 
+    const postLink = document.createElement('a') 
 
-    const postLinkDiv = document.createElement('div');
-    postLinkDiv.className += 'post-image-div';
+    const postLinkDiv = document.createElement('div') 
+    postLinkDiv.className += 'post-image-div' 
 
-    const postText = document.createElement('p');
-    postText.className += 'post-info';
-    postText.textContent = tittle;
+    const postText = document.createElement('p') 
+    postText.className += 'post-info' 
+    postText.textContent = tittle 
 
-    const postImg = document.createElement('img');
-    postImg.src = imgLink;
+    const postImg = document.createElement('img') 
+    postImg.src = imgLink 
 
-    postLinkDiv.appendChild(postText);
-    postLinkDiv.appendChild(postImg);
-    postLink.appendChild(postLinkDiv);
-    postElement.appendChild(postLink);
+    postLinkDiv.appendChild(postText) 
+    postLinkDiv.appendChild(postImg) 
+    postLink.appendChild(postLinkDiv) 
+    postElement.appendChild(postLink) 
 
-    return postElement;
+    return postElement 
   }
 
-
-  //Table headings
-
-  const tableRequest = new XMLHttpRequest();
-
-  tableRequest.onload = function () {
-    if (this.status === 200) {
-      try {
-        const tableObj = JSON.parse(this.responseText);
-
-        // Add table headings
-        const tableHead = document.querySelector('#table-head');
-        const tableRow = document.createElement('tr');
-
-        for (i = 0; i < tableObj.heading.length; i++) {
-          const tableElement = document.createElement('th');
-          tableElement.textContent = tableObj.heading[i];
-
-          tableRow.appendChild(tableElement);
-
-          if (i == tableObj.heading.length - 1) {
-            tableHead.appendChild(tableRow);
-          }
-        }
-      } catch {
-        console.warn('JSON not parsed');
-      }
-    } else {
-      console.warn('File not found');
+  // Search filter
```

Comment:
Wrap the search code as well in a function and call were menuLoader() is invoked.

Same for mobile-view listener

Reply:
Wrapped code into separate functions.